### PR TITLE
TM-65 re-add local port availability check

### PR DESCRIPTION
**Pull request by roastario**

### CODE REVIEW

**Review by josecoll — CHANGES_REQUESTED** [review]
can you remove all the redundant blank lines - otherwise looks good!

File: testing/node-driver/src/main/java/net/corda/testing/driver/SharedMemoryIncremental.java [hunk 1]
```diff
@@ -65,15 +69,28 @@ static private Unsafe getUnsafe() {
...
             }
-        } while (!UNSAFE.compareAndSwapLong(null, startingAddress, oldValue, newValue));
+            loopSuccess = isLocalPortAvailable(newValue) && UNSAFE.compareAndSwapLong(null, startingAddress, oldValue, newValue);
+        } while (!loopSuccess);
 
         return newValue.intValue();
     }
+
+
+    Boolean isLocalPortAvailable(Long portToTest) {
```
Comment (vkolomeyko) [inline testing/node-driver/src/main/java/net/corda/testing/driver/SharedMemoryIncremental.java]: There is already a method `net.corda.testing.internal.InternalTestUtilsKt#isLocalPortBound`, is it not possible to use it from Java?
(also implementation below does not seem to be calling `close()` on `ServerSocket`)

Comment (roastario) [reply testing/node-driver/src/main/java/net/corda/testing/driver/SharedMemoryIncremental.java]: we cannot access any kotlin code from this section :/ 

I've wrapped it in a try-with-resources block to ensure it's closed

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**